src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.cs

```diff
@@ -33,13 +34,25 @@ internal abstract partial class AbstractTypeImportCompletionService : ITypeImpor
         protected abstract string Language { get; }
 
         internal AbstractTypeImportCompletionService(Workspace workspace)
-            => CacheService = workspace.Services.GetRequiredService<IImportCompletionCacheService<CacheEntry, CacheEntry>>();
+        {
+            _workspace = workspace;
+            _workQueue = new(
+                   TimeSpan.FromSeconds(1),
+                   UpdateCacheAsync,
+                   EqualityComparer<ProjectId>.Default,
+                   AsynchronousOperationListenerProvider.NullListener,
+                   CancellationToken.None);
+
+            CacheService = _workspace.Services.GetRequiredService<IImportCompletionCacheService<CacheEntry, CacheEntry>>();
+        }
 
         public Task WarmUpCacheAsync(Project? project, CancellationToken cancellationToken)
         {
-            return project is null
-                ? Task.CompletedTask
-                : GetCacheEntriesAsync(project, forceCacheCreation: true, cancellationToken);
+            if (project is null)
+                return Task.CompletedTask;
+
+            _workQueue.AddWork(project.Id);
+            return _workQueue.WaitUntilCurrentBatchCompletesAsync();
```

Comment:
you definitely don't want to wait on the work to complete :)  that somewhat defeats the purpose.  you just want to put the item in the queue and then know that it will be taken care of later.

Reply:
This is the code path that returns the cache update task, not the code path for completion.

Reply:
could you point me to the code that looks at the return value of this method?